hr_switzerland/models/hr_employee.py

```diff
@@ -12,72 +12,76 @@
 
 from datetime import datetime
 
-from odoo import models, fields
+from odoo import models, fields, api
 
 
 class HrEmployee(models.Model):
     _inherit = 'hr.employee'
 
-    bonus_malus_hour = fields.Char(string="Bonus/Malus",
-                                   compute='_compute_bonus_malus_hour')
+    extra_hours_formatted = fields.Char(string="Balance",
+                                   compute='_compute_extra_hours')
 
     time_warning_balance = fields.Char(compute='_compute_time_warning_balance')
 
     time_warning_today = fields.Char(compute='_compute_time_warning_today')
 
-    bonus_malus_today = fields.Char(compute='_compute_bonus_malus_today')
+    extra_hours_today = fields.Char(compute='_compute_extra_hours_today')
 
     today_hour = fields.Char(compute='_compute_today_hour')
 
-    def _compute_bonus_malus_today(self):
-        for employee in self:
-            today_hour = employee._today_hour() - 8
+    today_hour_formatted = fields.Char(compute='_compute_today_hour_formatted')
 
-            employee.bonus_malus_today = '- ' if today_hour < 0 else '+'
-            employee.bonus_malus_today += employee.\
-                _convert_hour_to_time(today_hour)
+    @api.multi
+    @api.depends('today_hour')
+    def _compute_extra_hours_today(self):
+        for employee in self:
+            employee.extra_hours_today = \
+                '- ' if employee.today_hour < 0 else '+'
+            employee.extra_hours_today += employee.\
+                _convert_hour_to_time(employee.today_hour)
 
+    @api.multi
     def _compute_time_warning_balance(self):
         for employee in self:
-            if employee.bonus_malus < 0:
+            if employee.extra_hours < 0:
                 employee.time_warning_balance = 'red'
-            elif employee.bonus_malus >= 19:
+            elif employee.extra_hours >= 19:
                 employee.time_warning_balance = 'orange'
             else:
                 employee.time_warning_balance = 'green'
 
+    @api.multi
+    @api.depends('today_hour')
     def _compute_time_warning_today(self):
         for employee in self:
-            employee.time_warning_today = 'red' if employee._today_hour() < \
+            employee.time_warning_today = 'red' if employee.today_hour < \
                 8 else 'green'
 
+    @api.multi
     def _compute_today_hour(self):
         for employee in self:
-            employee.today_hour = employee._convert_hour_to_time(
-                employee._today_hour())
-
-    def _compute_bonus_malus_hour(self):
+            current_att_day = self.env['hr.attendance.day'].search([
+                ('employee_id', '=', employee.id),
+                ('date', '=', fields.Date.today())])
+            employee.today_hour = \
+                current_att_day.worked_hours - current_att_day.due_hours
```

Comment:
current_att_day.worked_hours doesn't return the current time elapsed, you need the previously method used for compute the employe.today_hour